File: src/main/java/uk/co/threebugs/mochiwhattotrade3/metatrader/Order.java
package uk.co.threebugs.mochiwhattotrade3.metatrader;

import lombok.Builder;
import lombok.Value;

import java.math.BigDecimal;

@Value
@Builder
public class Order {
    String symbol;
    String orderType;
    double lots;
    BigDecimal price;
    BigDecimal stopLoss;
    BigDecimal takeProfit;
    int magic;
    String comment;
    long expiration;
}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/metatrader/Order.kt
package uk.co.threebugs.mochiwhattotrade3.metatrader

import java.math.BigDecimal


data class Order(
        val symbol: String? = null,
        val orderType: String? = null,
        val lots: Double? = null,
        val price: BigDecimal? = null,
        val stopLoss: BigDecimal? = null,
        val takeProfit: BigDecimal? = null,
        val magic: Int? = null,
        val expiration: Long? = null,
        val comment: String? = null
)